temporalcli/commands.operator_namespace.go

```diff
@@ -0,0 +1,335 @@
+		OwnerEmail:                       c.Email,
+		WorkflowExecutionRetentionPeriod: durationpb.New(retention),
+		Clusters:                         clusters,
+		ActiveClusterName:                c.ActiveCluster,
+		Data:                             data,
+		IsGlobalNamespace:                c.Global,
+		HistoryArchivalState:             archivalState(c.HistoryArchivalState.Value),
+		HistoryArchivalUri:               c.HistoryUri,
+		VisibilityArchivalState:          archivalState(c.VisibilityArchivalState.Value),
+		VisibilityArchivalUri:            c.VisibilityUri,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to create namespace: %w", err)
+	}
+	cctx.Printer.Println(color.GreenString("Namespace %s successfully registered.", nsName))
+	return nil
+}
+
+func (c *TemporalOperatorNamespaceDeleteCommand) run(cctx *CommandContext, args []string) error {
+	nsName := args[0]
+	yes, err := cctx.promptString(
+		color.RedString("Are you sure you want to delete namespace %s? Type namespace name to confirm:", nsName),
+		nsName,
+		c.Yes)
+	if err != nil {
+		return err
+	}
+
+	if !yes {
+		return fmt.Errorf("user denied confirmation")
+	}
+
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	resp, err := cl.OperatorService().DeleteNamespace(cctx, &operatorservice.DeleteNamespaceRequest{
+		Namespace: nsName,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to delete namespace: %w", err)
+	}
+
+	if cctx.JSONOutput {
+		return cctx.Printer.PrintStructured(resp, printer.StructuredOptions{})
+	}
+	cctx.Printer.Println(color.GreenString("Namespace %s has been deleted.", nsName))
+	return nil
+}
+
+func (c *TemporalOperatorNamespaceDescribeCommand) run(cctx *CommandContext, args []string) error {
+	nsName, nsID, err := getNamespaceFromIDArgs(cctx, c.NamespaceId, args)
+	if err != nil {
+		return err
+	}
+
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	resp, err := cl.WorkflowService().DescribeNamespace(cctx, &workflowservice.DescribeNamespaceRequest{
+		Namespace: nsName,
+		Id:        nsID,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to describe namespace: %w", err)
+	}
+
+	if cctx.JSONOutput {
+		return cctx.Printer.PrintStructured(resp, printer.StructuredOptions{})
+	}
+	return printNamespaces(cctx, resp)
+}
+
+func (c *TemporalOperatorNamespaceListCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+
+	var nextPageToken []byte
+	for {
+		resp, err := cl.WorkflowService().ListNamespaces(cctx, &workflowservice.ListNamespacesRequest{
+			NextPageToken: nextPageToken,
+			PageSize:      100,
+		})
+		if err != nil {
+			return fmt.Errorf("failed listing namespaces: %w", err)
+		}
+
+		if cctx.JSONOutput {
+			// For JSON we are going to dump one line of JSON per execution
```

Comment:
Hrmm, I wonder if this is needed as a lazy stream, but it is fine. I have added this to #448.